smarts/core/sumo_road_network.py

```diff
@@ -324,10 +324,15 @@ def oncoming_lanes_at_offset(self, offset: float) -> List[RoadMap.Lane]:
             if not nearby_lanes:
                 return result
             my_vect = self.vector_at_offset(offset)
-            my_norm = np.linalg.norm(my_vect)
+            my_norm = max(np.linalg.norm(my_vect), 1e-6)
             threshold = -0.995562  # cos(175*pi/180)
             for lane, _ in nearby_lanes:
-                lv = lane.vector_at_offset(offset)
+                if lane == self:
```

Comment:
Why does `nearby_lanes` have `self` in it? It feels counterintuitive.

Reply:
```python
 nearby_lanes = self._map.nearest_lanes(pt, radius=radius)
```
nearby_lanes is obtained from `nearest_lanes` method which is a method of parent `RoadMap` class. So it doesn't know about the current lane and hence can return the current lane `self` as one of the nearby lanes to this point.